client/webserver/site/src/js/markets.ts

```diff
@@ -598,22 +599,34 @@ export default class MarketsPage extends BasePage {
 
   /* setHighLow calculates the high and low rates over the last 24 hours. */
   setHighLow () {
-    const cache = this.market?.candleCaches[fiveMinBinKey]
-    if (!cache) {
-      for (const s of this.stats) {
-        s.tmpl.high.textContent = '-'
-        s.tmpl.low.textContent = '-'
+    let [high, low] = [0, 0]
+    const spot = this.market.cfg.spot
+    // Use spot values for 24 hours high and low rates if it is available. We
+    // will default to setting it from candles if it's not.
+    if (spot && spot.low24 && spot.high24) {
+      high = spot.high24
+      low = spot.low24
+    } else {
+      const cache = this.market?.candleCaches[fiveMinBinKey]
+      if (!cache) {
+        if (this.candleDur !== fiveMinBinKey) {
+          this.candleDur = fiveMinBinKey
+          this.requestCandles()
+          return
+        }
+        for (const s of this.stats) {
+          s.tmpl.high.textContent = '-'
+          s.tmpl.low.textContent = '-'
+        }
+        return
       }
-      return
-    }
 
-    // We'll eventually get this data in the spot, but for now, we must set it
-    // from candles.
-    let [high, low] = [0, 0]
-    for (let i = cache.candles.length - 1; i >= 0; i--) {
-      const c = cache.candles[i]
-      if (low === 0 || (c.lowRate > 0 && c.lowRate < low)) low = c.lowRate
-      if (c.highRate > high) high = c.highRate
+      // Set high and low rates from candles.
+      for (let i = cache.candles.length - 1; i >= 0; i--) {
+        const c = cache.candles[i]
+        if (low === 0 || (c.lowRate > 0 && c.lowRate < low)) low = c.lowRate
+        if (c.highRate > high) high = c.highRate
+      }
```

Comment:
Not from this work, but there seems to be something missing here. Don't we need something like

```Typescript
const aDayAgo = new Date().getTime() - 86400000
for (let i = cache.candles.length - 1; i >= 0; i--) {
  const c = cache.candles[i]
  if (c.endStamp < aDayAgo) break
  ...
}
```

Reply:
Good catch. I think we do.